File: api/eggprices.py
from flask import Flask, jsonify
import requests
from bs4 import BeautifulSoup

app = Flask(__name__)

@app.route('/api/eggprices', methods=['GET'])
def get_egg_prices():
    url = 'https://e2necc.com/home/eggprice'
    response = requests.get(url)

    if response.status_code == 200:
        soup = BeautifulSoup(response.content, 'html.parser')
        table = soup.find('table', {'border': '1px'})

        headers = [header.text for header in table.find_all('th')]
        rows = []
        for row in table.find_all('tr')[2:]:
            cells = row.find_all('td')
            cells = [cell.text.strip() for cell in cells]
            rows.append(cells)

        data = {
            'headers': headers,
            'rows': rows
        }
        return jsonify(data)
    else:
        return jsonify({'error': 'Failed to retrieve the page'}), response.status_code

# Vercel requires the app to be exported as a module-level variable named "app"
if __name__ != '__main__':
    app = app

File: public/php/eggprices.php
<?php
header('Content-Type: application/json');

$url = 'https://e2necc.com/home/eggprice';
$html = file_get_contents($url);

if ($html !== false) {
    $dom = new DOMDocument();
    @$dom->loadHTML($html);
    $xpath = new DOMXPath($dom);

    $table = $xpath->query('//table[@border="1px"]')->item(0);
    if ($table) {
        $headers = [];
        $rows = [];

        // Get table headers
        $headerNodes = $xpath->query('.//th', $table);
        foreach ($headerNodes as $headerNode) {
            $headers[] = trim($headerNode->textContent);
        }

        // Get table rows
        $rowNodes = $xpath->query('.//tr', $table);
        foreach ($rowNodes as $rowIndex => $rowNode) {
            if ($rowIndex < 2) continue; // Skip the first two rows
            $cells = [];
            $cellNodes = $xpath->query('.//td', $rowNode);
            foreach ($cellNodes as $cellNode) {
                $cells[] = trim($cellNode->textContent);
            }
            if (!empty($cells)) {
                $rows[] = $cells;
            }
        }

        $data = [
            'headers' => $headers,
            'rows' => $rows
        ];
        echo json_encode($data);
    } else {
        echo json_encode(['error' => 'Failed to find the table']);
    }
} else {
    echo json_encode(['error' => 'Failed to retrieve the page']);
}
?>